src/main/scala/is/hail/methods/MaximalIndependentSet.scala

```diff
@@ -8,11 +8,30 @@ import scala.reflect.ClassTag
 
 object MaximalIndependentSet {
 
-  def apply[VD: ClassTag, ED: ClassTag](g: Graph[VD, ED], undirected: Boolean = false): Set[Long] = {
+  def apply[VD: ClassTag, ED: ClassTag](g: Graph[VD, ED], optComparisonFunction: Option[(Int, Int) => Int] = None): Set[Long] = {
 
+    val comparisonFunction: (Int, Int) => Int = optComparisonFunction.getOrElse((_, _) => 0)
     type Message = (Int, VertexId)
-    val pairOrd = implicitly[Ordering[Message]]
-    import pairOrd.mkOrderingOps
+
+    //val subOrd = Ordering.by[Message, Int]
+    //val ord: Ordering[Message] = Ordering.by[Message, Int](_._1).thenComparing(Ordering.by[Message, Int](_._2.toInt))
+    val ord = new Ordering[Message] {
+      override def compare(x: (Int, VertexId), y: (Int, VertexId)): Int = {
+        //By degree
+        (x._1 compare y._1) match {
+          case 0 => {
+            //By function
+            comparisonFunction(x._2.toInt, y._2.toInt) match {
+              //By VertexID
+              case 0 => x._2 compare y._2
+              case c => c
+            }
+          }
+          case c => c
+        }
+      }
+    }
+    import ord.mkOrderingOps
```

Comment:
Yeah wow, the `Ordering` class has no composition operator.

Let's add this to utils:

```scala
def dictionaryOrdering[T](ords: Ordering[T]*): Ordering[T] = {
  new Ordering[T] {
    def compare(x: T, y: T): Int = {
      var i = 0
      while (i < ords.size) {
        val v = ords(i).compare(x, y)
        if (v != 0)
          return v
        i += 1
      }
      return 0
    }
  }
}
```

and change this to:

```scala
val cmp = optComparisonFunction.getOrElse((_,_) => 0)
val cmpOrd = new Ordering[Int] { def compare(x: Int, y: Int): Int = cmp(x, y) }
val ord = dictionaryOrdering[Message](
  Ordering.by(_._1),
  Ordering.by[Message, Int](_._1)(cmpOrd),
  Ordering.by(_._2))
```